product_docs/docs/epas/15/epas_guide/03_database_administration/01_configuration_parameters/01_setting_new_parameters.mdx

```diff
@@ -87,3 +87,81 @@ application_name=psql
 2020-05-08 10:42:17 IST DETAIL:  Connection matched pg_hba.conf line 84:
 "local   all             all               md5"
 ```
+
+### Obfuscating the LDAP password
+
+You can use LDAP authentication to authenticate database users. For more information , see [LDAP authentication](https://www.postgresql.org/docs/15/auth-ldap.html).
+
+The `ldapbindpasswd` is stored in plaintext in the pg_hba.conf file. It is possible to store an encrypted `ldapbindpasswd` by writing an extension to decrypt. This extension should contain a decryption function with the C signature. You can add this extension to the `shared_preload_libraries` parameter in postgresql.conf file. You can restart your server to load the changes in this parameter.
+
+```ini
+shared_preload_libraries = '$libdir/ldap_password_func'
+```
+
+This example shows the code of extension `ldap_password_func.c` with the function `rot13_passphrase` to decrypt the password:
+
+```c
+/*-------------------------------------------------------------------------
+ *
+ * ldap_password_func.c
+ *
+ * Loadable PostgreSQL module to mutate the ldapbindpasswd. This
+ * implementation just hands back the configured password rot13'd.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include <float.h>
+#include <stdio.h>
+
+#include "libpq/libpq.h"
+#include "libpq/libpq-be.h"
+#include "libpq/auth.h"
+#include "utils/guc.h"
+
+PG_MODULE_MAGIC;
+
+void		_PG_init(void);
+void		_PG_fini(void);
+
+/* hook function */
+static char*	rot13_passphrase(char *password);
+
+/*
+ * Module load callback
+ */
+void
+_PG_init(void)
+{
+	ldap_password_hook = rot13_passphrase;
+}
+
+void
+_PG_fini(void)
+{
+	/* do  nothing yet */
+}
+
+static char*
+rot13_passphrase(char *pw)
+{
+	size_t size = strlen(pw) + 1;
+
+	char* new_pw = (char*) palloc(size);
+	strlcpy(new_pw, pw, size);
+	for (char *p = new_pw; *p; p++)
+	{
+		char		c = *p;
+
+		if ((c >= 'a' && c <= 'm') || (c >= 'A' && c <= 'M'))
+			*p = c + 13;
+		else if ((c >= 'n' && c <= 'z') || (c >= 'N' && c <= 'Z'))
+			*p = c - 13;
+	}
+
+	return new_pw;
+}
+```
```

Comment:
1. Add the extension to the `shared_preload_libraries` parameter in the postgresql.conf file.
1. Restart your server.